### PR TITLE
Ensure that crypto allocations are using BufferSource

### PR DESCRIPTION
**Pull request by jasnell**
... ensures that all ArrayBuffer instances are using BufferSource rather than external allocations.

### CODE REVIEW

**Comment by jasnell** [comment]
A handful of the remaining uses of `kj::heapArray` could actually be converted to `KJ_STACK_ARRAY` but will handle that cleanup in a separate PR.

**Review by anonrig — APPROVED** [review]

File: src/workerd/api/crypto/crypto.h [hunk 1]
```diff
@@ -207,19 +207,28 @@ class CryptoKey: public jsg::Object {
       KJ_SWITCH_ONEOF(publicExponent) {
         KJ_CASE_ONEOF(array, BigInteger) {
           if (fixPublicExp) {
-            auto expCopy = kj::heapArray<kj::byte>(array.asPtr());
-            jsg::BackingStore expBack = jsg::BackingStore::from(kj::mv(expCopy));
+            // alloc will, by default create a Uint8Array
```
Comment (erikcorry) [inline src/workerd/api/crypto/crypto.h:210]: If the comment is intended to document what type is hiding behind `auto` then would it not be nicer to just write the type there.

If that's not what is intended, then the reason for the comment is unclear.

File: src/workerd/api/crypto/crypto.c++ [hunk 2]
```diff
@@ -538,18 +542,17 @@ jsg::Promise<jsg::Ref<CryptoKey>> SubtleCrypto::unwrapKey(jsg::Lock& js,
 
     validateOperation(unwrappingKey, normalizedAlgorithm.name, CryptoKeyUsageSet::unwrapKey());
 
-    kj::Array<kj::byte> bytes =
-        unwrappingKey.impl->unwrapKey(kj::mv(normalizedAlgorithm), wrappedKey);
+    auto bytes = unwrappingKey.impl->unwrapKey(js, kj::mv(normalizedAlgorithm), wrappedKey);
```
Comment (erikcorry) [inline src/workerd/api/crypto/crypto.c++:545]: Not really a comment for this PR, but I wish we had less auto and more explicit types, and here you are going the other way.  For the reader it's just less clear.  Could be that the IDE tells you the type, but we don't always have that (eg here in the code review).

**Review by erikcorry — APPROVED** [review]
This is great, should help with the sandboxing of V8.

File: src/workerd/api/crypto/crypto.c++ (hunk 2 above)
Comment (jasnell) [reply src/workerd/api/crypto/crypto.c++:545]: Noted ;-) .... I think we've found the c++ equivalent to spaces vs. tabs tho lol ... some folks seem to really like `auto` (like myself) and others really hate it

**Review by Pjrich1313 — COMMENTED** [review]
Rerun

**Pull request merged**